lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -78,29 +78,38 @@ def figure_edit(axes, parent=None):
                 continue
             linedict[label] = line
         curves = []
-        linestyles = list(six.iteritems(LINESTYLES))
-        drawstyles = list(six.iteritems(DRAWSTYLES))
-        markers = list(six.iteritems(MARKERS))
+
+        def prepare_data(d, init):
+            """Prepare entry for FormLayout.
+            """
+            # List items in dict, dropping duplicate values, sorting by values.
+            kvs = [(k, v) for v, k in
+                   sorted({v: k for k, v in d.items()}.items())]
```

Comment:
Pretty sure that this will not be deterministic as the iteration order of dictionaries is random process to process in 3.4+.  Dict comprehensions are the same as a series of `d[k]=v ` so the same override semantics apply.

This might be better done as

```python
dd = defaultdict(list)
for k, v in d.items():
    dd[v].append(k)
Uniqed = {k: v[0] for k, v in dd.items()}
```